extensions/data-transfer/portability-data-transfer-backblaze/src/main/java/org/datatransferproject/datatransfer/backblaze/videos/BackblazeVideosImporter.java

```diff
@@ -78,11 +79,16 @@ public ImportResult importItem(
 
   private String importSingleVideo(BackblazeDataTransferClient b2Client, VideoModel video)
       throws IOException {
-    InputStream videoFileStream =
-        imageStreamProvider.getConnection(video.getContentUrl().toString()).getInputStream();
+    try (InputStream videoFileStream =
+        imageStreamProvider.getConnection(video.getContentUrl().toString()).getInputStream()) {
 
-    return b2Client.uploadFile(
-        String.format("%s/%s.mp4", VIDEO_TRANSFER_MAIN_FOLDER, video.getDataId()),
-        jobStore.getTempFileFromInputStream(videoFileStream, video.getDataId(), ".mp4"));
+      return b2Client.uploadFile(
+          String.format("%s/%s.mp4", VIDEO_TRANSFER_MAIN_FOLDER, video.getDataId()),
+          jobStore.getTempFileFromInputStream(videoFileStream, video.getDataId(), ".mp4"));
+    } catch (FileNotFoundException e) {
+      monitor.severe(
```

Comment:
(just to refresh my memory), how does this behave with the idempotent executor? Would we want to transform this execption into an IOException so it will get swallowed and recorded? (in what other ways would this get marked as a "failure" in the job record?

Reply:
So for this the video has actually already been deleted so we don't mark this as a failure. With our work to add more details to transfers we plan to add a new state where we can mark an item specifically as skipped and will apply that to this use case but for now this is a very rare edge case where an item has just been deleted so we skip without marking the job as failed.